### ISSUE TITLE
Ability to return a IAsyncOperation with a value

### ISSUE THREAD

**Issue by Jaiganeshkumaran** [microsoft/cppwinrt#1036]
Today if you want to return an IAsyncOperation with a specific value, you either need to use a coroutine and implement IAsyncOperation yourself. However in some cases you might just want to return a value synchronously as if you returned the type itself but use an IAsyncOperation.

**Comment by Jaiganeshkumaran** [comment]
> Sure, you can do that (what @sylveon also suggested) but that doesn't get away from the fact that a new async object is allocated, which is precisely what a coroutine will do for you.

That's right that it will allocate a new async object however in some othercase it will return an existing object and only return a new object with the existing method can't be used or doesn't work. Something like .NET's Task.FromResult method. A coroutine can't have both return and co_return, it must have a co_return otherwise it can't be a coroutine.

**Comment by kennykerr** [comment]
Yep, I just don't see this as a very common scenario nor the cost of coroutines so prohibitive to necessitate this. But you're certainly free to do as @sylveon suggested and write a simple implementation of `IAsyncOperation` that does just that. If you then find it works remarkably well you can always come back here with that experience and we can reconsider including something like that more directly.

**Comment by sylveon** [comment]
FWIW Task.FromResult in C# allocates. Using a ValueTask wouldn't in the case of a non-async return but value tasks have several other pitfalls.
